packages/frontend/src/features/semanticViewer/runners/SemanticViewerResultsRunner.ts

```diff
@@ -86,6 +90,85 @@ export class SemanticViewerResultsRunner extends ResultsRunner {
         this.fields = fields;
     }
 
+    pivotChartOptions(): {
+        indexLayoutOptions: VizIndexLayoutOptions[];
+        valuesLayoutOptions: VizValuesLayoutOptions[];
+        pivotLayoutOptions: VizPivotLayoutOptions[];
+    } {
+        return {
+            indexLayoutOptions: this.columns.reduce((acc, column) => {
+                const field =
+                    SemanticViewerResultsRunner.findSemanticLayerFieldFromColumn(
+                        this.fields,
+                        column.reference,
+                    );
+                if (field?.kind === FieldType.DIMENSION) {
+                    acc.push({
+                        reference: column.reference,
+                        type: this.getAxisType(column),
+                    });
+                }
+                return acc;
+            }, [] as VizIndexLayoutOptions[]),
+            valuesLayoutOptions: this.columns.reduce((acc, column) => {
+                const field =
+                    SemanticViewerResultsRunner.findSemanticLayerFieldFromColumn(
+                        this.fields,
+                        column.reference,
+                    );
+                if (field?.kind === FieldType.METRIC) {
+                    acc.push({
+                        reference: column.reference,
+                    });
+                }
+                return acc;
+            }, [] as VizValuesLayoutOptions[]),
+            pivotLayoutOptions: this.columns.filter((column) => {
+                const field =
+                    SemanticViewerResultsRunner.findSemanticLayerFieldFromColumn(
+                        this.fields,
+                        column.reference,
+                    );
+                return field?.kind === FieldType.DIMENSION;
+            }),
+        };
+    }
+
+    defaultPivotChartLayout(): VizChartLayout | undefined {
+        let xColumn: VizColumn | undefined;
+        let yColumn: VizColumn | undefined;
+
+        this.columns.forEach((column) => {
+            const field =
+                SemanticViewerResultsRunner.findSemanticLayerFieldFromColumn(
+                    this.fields,
+                    column.reference,
+                );
+            if (field?.kind === FieldType.DIMENSION) {
+                xColumn = column;
+            } else if (field?.kind === FieldType.METRIC) {
+                yColumn = column;
+            }
+        });
```

Comment:
This loops through all columns and will still try to get the field for each of those - setting x or y to be the last item in the columns list depending on type

was thinking of something very similar to what we had prior to the change we did during the call, e.g.
```suggestion
        const xColumn = this.columns.find((column) => {
            const field =
                SemanticViewerResultsRunner.findSemanticLayerFieldFromColumn(
                    this.fields,
                    column.reference,
                );

             return field?.kind === FieldType.DIMENSION;
        });
        
        const yColumn = this.columns.find((column) => {
            const field =
                SemanticViewerResultsRunner.findSemanticLayerFieldFromColumn(
                    this.fields,
                    column.reference,
                );

             return field?.kind === FieldType.METRIC;
        });
```